core/core_bind.cpp

```diff
@@ -390,7 +390,7 @@ int64_t _OS::get_unix_time_from_datetime(Dictionary datetime) const {
 	unsigned int hour = ((datetime.has(HOUR_KEY)) ? static_cast<unsigned int>(datetime[HOUR_KEY]) : 0);
 	unsigned int day = ((datetime.has(DAY_KEY)) ? static_cast<unsigned int>(datetime[DAY_KEY]) : 1);
 	unsigned int month = ((datetime.has(MONTH_KEY)) ? static_cast<unsigned int>(datetime[MONTH_KEY]) : 1);
-	unsigned int year = ((datetime.has(YEAR_KEY)) ? static_cast<unsigned int>(datetime[YEAR_KEY]) : 0);
+	unsigned int year = ((datetime.has(YEAR_KEY)) ? static_cast<unsigned int>(datetime[YEAR_KEY]) : 1970);
```

Comment:
There will be still freeze, when user will set `datetime[YEAR_KEY]` to 0.
Probably better is to validate user data(don't know if current algorithm allow to return negative unix time)

Reply:
Negative times (years before `1970`) seem to work fine down to `1`, but indeed `0` and negative years also freeze. I'll had an err cond, someone should clean up the algorithm and make it actually robust if we want to handle BC years.

Reply:
I don't think we can even say it works that far. I don't see a special case for October 1582, so I assume it doesn't work properly for times before then.

Reply:
Yeah this needs a complete overhaul, or using a third-party library or possibly modern OS APIs.

For now I'm just handling the most obvious crash. BTW negative years would overflow as it's being cast to `unsigned int`, so those might not trigger an infinite loop, just a veeeeeeeeeeeery long one when iterating all the way to `UINT64_MAX` :P

Reply:
After further investigation, for the sake of sanity of programmers, most tools use the Proleptic Gregorian calendar which extends the Gregorian calendar back before 1582 as if it had always been adopted. With the Proleptic Gregorian calendar, we don't need a special case for October 1582. https://en.wikipedia.org/wiki/Proleptic_Gregorian_calendar

However, if Godot wishes to support historical dates, a parallel API that handles the Julian calendar would be nice.